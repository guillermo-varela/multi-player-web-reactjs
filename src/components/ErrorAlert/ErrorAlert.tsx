import React from 'react';
import Alert from 'react-bootstrap/Alert';
import { FormattedMessage } from 'react-intl';

type Message = {
  message?: String;
}

const ErrorAlert: React.FC<Message> = ({ message }) => {
  return (
    <Alert variant="danger">
      {message ?
        message
        :
        <span>
          <FormattedMessage id='component.error.message' />
        </span>
      }

    </Alert>
  );
}

export default ErrorAlert;
