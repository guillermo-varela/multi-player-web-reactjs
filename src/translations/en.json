{
  "component.about.code": "The source code for this project can be found at:",
  "component.about.codeSplit": "Route based code splitting",
  "component.about.csrf": "Use of custom HTTP headers in stateless/sessionless environments to prevent CSRF attacks",
  "component.about.forms.formik": "Forms in React using Formik",
  "component.about.hello": "Hi again :)",
  "component.about.internationalization": "Support for multiple languages (internationalization) using react-intl",
  "component.about.introduction": "This web application was developed to learn about:",
  "component.about.jwt": "Stateless/sessionless authentication on a server using \"Javascript Object Signing and Encryption\" (JOSE) and \"JSON Web Token\" (JWT)\"",
  "component.about.openAPI": "OpenAPI 3",
  "component.about.pixabay": "The multimedia content was fetched using the {link}.",
  "component.about.pixabay.name": "Pixabay API",
  "component.about.redux.typescript": "Redux using TypeScript",
  "component.about.rest.webflux": "REST services with reactive architecture (non-blocking) with Spring WebFlux and Spring Security",
  "component.error.message": "Ops! An error occurred...",
  "component.signin.button.sending": "Sending...",
  "component.signin.button.submit": "Sign In",
  "component.signin.credentials.hint": "Use user@test.com as email and password to test",
  "component.signin.empty.email": "Enter an email",
  "component.signin.incorrect.credentials": "Incorrect credentials",
  "component.signin.invalid.email": "Enter a valid email",
  "component.signin.invalid.password": "Enter a password",
  "component.signin.label.email": "Email",
  "component.signin.label.password": "Password",
  "menu.about": "About",
  "menu.home": "Home",
  "menu.images": "Images",
  "menu.more": "View More",
  "menu.sections": "Sections",
  "menu.signIn": "Sign In",
  "menu.signOut": "Sign Out",
  "menu.signOut.error.message": "Sign out failed",
  "menu.signOut.error.title": "Ops!",
  "menu.videos": "Videos",
  "multimedia.source": "Content from ",
  "multimedia.type.image": "Image",
  "multimedia.type.video": "Video",
  "nature.hello": "Hello!",
  "nature.introduction": "On this site you can find multimedia content about nature.",
  "nature.multimedia.endOfContent": "That's all for now...",
  "nature.title": "Nature",
  "root.loading": "Loading...",
  "root.locale.label": "Language",
  "root.nature.description": "Images and videos of flora and fauna.",
  "root.space.description": "Images and videos of the universe.",
  "space.hello": "Hello!",
  "space.images.introduction": "Multiple images related to outer space.",
  "space.introduction": "On this site you will find multimedia content about outer space.",
  "space.menu.title": "Space",
  "space.title": "Outer Space",
  "space.videos.introduction": "Mind-blowing videos from outer space."
}
