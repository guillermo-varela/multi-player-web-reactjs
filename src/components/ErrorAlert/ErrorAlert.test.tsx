import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Alert from 'react-bootstrap/Alert';
import { IntlProvider } from 'react-intl';
import en from '../../translations/en.json';
import ErrorAlert from './ErrorAlert';

describe('<ErrorAlert />', () => {
  it('renders <Alert />', () => {
    const wrapper = mount(buildComponent());
    expect(wrapper.find(Alert)).toHaveLength(1);
  });

  it('renders <Alert /> with an error message', () => {
    const alertWrapper = mount(buildComponent()).find(Alert);
    expect(alertWrapper.childAt(0).text()).toBe(en['component.error.message']);
  });

  it('renders <Alert /> with a custom error message', () => {
    const alertWrapper = mount(buildComponent('this is a test')).find(Alert);
    expect(alertWrapper.childAt(0).text()).toBe('this is a test');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(buildComponent()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const buildComponent = (message?: string) => {
    return (
      <IntlProvider locale='en' messages={en}>
        <ErrorAlert message={message} />
      </IntlProvider>
    );
  };
});
