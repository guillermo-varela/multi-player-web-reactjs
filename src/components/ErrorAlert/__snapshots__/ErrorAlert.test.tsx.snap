// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAlert /> matches snapshot 1`] = `
<div
  className="fade alert alert-danger show"
  role="alert"
>
  <span>
    Ops! An error occurred...
  </span>
</div>
`;
